src/libs/actions/OnyxUpdateManager/utils/GetMissingOnyxUpdatesPromise.ts

```diff
@@ -0,0 +1,6 @@
+import type {Response} from '@src/types/onyx';
+import createProxyForObject from '@src/utils/createProxyForObject';
+
+const GetMissingOnyxUpdatesPromiseValue = {GetMissingOnyxUpdatesPromise: undefined as Promise<Response | Response[] | void> | undefined};
+
+export default createProxyForObject(GetMissingOnyxUpdatesPromiseValue);
```

Comment:
As you might guess I'm not a big fan of this proxy object pattern 😄. These feel more or less like global variables and are prone to error in the same way since their values can be changed anywhere, at anytime and in any way.

It feels like we're pretty much creating a [lock](https://en.wikipedia.org/wiki/Lock_(computer_science)) with this object so it would be nice if we could abstract it like one. However, I'm not sure exactly how it would look.

Cleaning this up is NAB but if you know an easy way to write it I'm curious.